docs/articles/documentation/test-api/assertions/index.md

```diff
@@ -0,0 +1,108 @@
+which accepts `actual` value argument. Then the assertion method follows. An assertion method accepts expected a value
+and optionally other arguments. E.g. deep equality assertion:
+
+```text
+await t.expect( actual ).eql( expected, message, options );
+```
+
+The sample below demonstrates how to use assertions:
+
+```js
+import { Selector } from 'testcafe';
+
+
+fixture `Example page`
+   .page `http://devexpress.github.io/testcafe/example/`;
+
+
+test('Check property of element', async t => {
+   const developerNameInput = Selector('#developer-name');
+
+   await t
+       .expect(developerNameInput.value).eql('', 'input is empty')
+       .typeText(developerNameInput, 'Peter')
+       .expect(developerNameInput.value).eql('Peter', 'input has text');
+});
+```
+
+## Smart Assertion Query Mechanism
+
+In synchronous functional testing, you can perform the required assertions immediately after test action was executed.
+
+![Synchronous Functional Testing](../../../images/report.png)
+
+In web, functional tests are asynchronous. This means that we can not get the expected changes immediately after an end-user action.
+For example, the tested web page can send a request to the server for the required data, and this can take some time;
+or an end-user action launches animation after which the web page will reach its final state.
+All of these time gaps can not be pre-calculated, because they depend on various factors: computer performance,
+network connection speed, etc. In this case, if we perform assertions immediately after the test action finished,
+we can get an indefinite result.
+
+![Asynchronous Functional Testing](../../../images/report.png)
+
+To perform asynchronous functional tests, the additional timeout as usually added.
+
+![Asynchronous Functional Testing with Extra Waiting](../../../images/report.png)
+
+To make such tests stable, you need to add a timeout that will guarantee you a successful completion of required changes.
+Adding such timeouts can increase the test running time because of extra waiting.
+
+The main feature of TestCafe assertions is the use of built-in assertion query mechanism.
+To watch the state of web page elements, TestCafe provides [Selectors](../selecting-page-elements/selectors.md).
+If the TestCafe assertion receives a [Selector's property](../selecting-page-elements/selectors.md#snapshot-api-shorthands)
+as an actual value, TestCafe uses the smart assertion query mechanism:
+if an assertion did not passed, the test does not fail immediately. The assertion retries to pass multiple times and
+each time it re-requests the actual shorthand value. The test fails if the assertion could not complete successfully
+within a timeout.
+
+![TestCafe Smart Assertion Query Mechanism](../../../images/report.png)
+
+The approach described above allows you to create stable tests that lack random errors.
+Using this approach, you can decrease the amount of time required to run all your tests without extra waitings.
```

Comment:
`The approach described above`, `Using this approach`. Rephrase into single sentence without awkward `this/that approach` references